include/quda_matrix.h

```diff
@@ -671,91 +671,29 @@ namespace quda {
 
 
   template<class T, int N>
-    __device__ __host__ inline
-    void setIdentity(Matrix<T,N>* m){
-
-#pragma unroll
-      for (int i=0; i<N; ++i){
-        (*m)(i,i) = 1;
-#pragma unroll
-        for (int j=i+1; j<N; ++j){
-          (*m)(i,j) = (*m)(j,i) = 0;
-        }
-      }
-    }
-
-
-  template<int N>
-    __device__ __host__ inline
-    void setIdentity(Matrix<float2,N>* m){
-
-#pragma unroll
-      for (int i=0; i<N; ++i){
-        (*m)(i,i) = make_float2(1,0);
-#pragma unroll
-        for (int j=i+1; j<N; ++j){
-          (*m)(i,j) = (*m)(j,i) = make_float2(0.,0.);
-        }
-      }
-    }
-
-
-  template<int N>
-    __device__ __host__ inline
-    void setIdentity(Matrix<double2,N>* m){
-
+   __device__ __host__ inline void setIdentity(Matrix<T,N>* m)
+  {
 #pragma unroll
-      for (int i=0; i<N; ++i){
-        (*m)(i,i) = make_double2(1,0);
+    for (int i=0; i<N; ++i){
+      (*m)(i,i) = 1;
```

Comment:
I'm surprised this works in all cases; in particular, I don't think you can initialize a `std::complex<float/double>` with an integer... but I guess we'll cross that bridge if/when we get there.

Reply:
This isn't new code, I just deleted some unused routines.  But note that we use `quda::complex` not `std::complex`.